src/main/webapp/js/searchableField.js

```diff
@@ -10,35 +29,59 @@
 Behaviour.specify(".searchable", 'searchableField', 200, function(el) {
     var results = {};
 
-    new ComboBox(el, function(query) {
+    /**
+     * This combobox comes from https://github.com/jenkinsci/jenkins/blob/master/war/src/main/webapp/scripts/combobox.js
+     */
+    var combobox = new ComboBox(el, function(query) {
         return Object.keys(results);
     }, {});
 
-    el.addEventListener('input', function(e) {
+    el.addEventListener('input', debounce(function (e) {
+        // Clear fields dependent on this
+        document.querySelectorAll('[filldependson~="' + e.target.name.replace("_.", "") + '"]').forEach(function (el) {
+            if (el.name !== e.target.name) {
+                el.value = "";
+                document.getElementById(el.getAttribute('valueField')).value = "";
+            }
+        });
+        document.getElementById(el.getAttribute('valueField')).value = "";
+
+        if (el.value.length < 2) { // Only perform a search if there are enough characters
+            results = {};
+            return;
+        }
+
+        // Get the values of the field this depends on
         var parameters = (el.getAttribute("fillDependsOn") || "").split(" ")
-            .reduce(function(params, fieldName) {
+            .reduce(function (params, fieldName) {
                 var dependentField = findNearBy(el, fieldName);
                 if (dependentField) {
                     params[fieldName] = dependentField.value;
                 }
                 return params;
             }, {});
+        // Request the search results
         new Ajax.Request(el.getAttribute("fillUrl"), {
             parameters: parameters,
-            onSuccess: function(rsp) {
+            onSuccess: function (rsp) {
                 var valueIdentifier = el.getAttribute("valueIdentifier");
                 results = (rsp.responseJSON.data && rsp.responseJSON.data.values || [])
                     .reduce(function (flattened, result) {
                         flattened[result.name] = result[valueIdentifier];
                         return flattened;
                     }, {});
+            },
+            onError: function (rsp) {
```

Comment:
Based on the awful documentation at http://api.prototypejs.org/ajax/Ajax/Request/, this should be `onFailure`